astroplan/utils.py

```diff
@@ -2,9 +2,60 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-__all__ = []
+from .exceptions import OldEarthRotationDataWarning
+from astropy.time import Time
+import astropy.units as u
+from astropy.utils.data import get_pkg_data_filename, download_file
+from astropy.utils import iers
+import os
 
-# Nothing here for now.
-# We could remove this file, but usually sooner or later there
-# will be some `utils` helper functions that could go here,
-# so keeping it for now.
+try:
+    # Python 3
+    from urllib.error import URLError
+except ImportError:
+    # Python 2
+    from urllib2 import URLError as URLError
+
+import warnings
+
+__all__ = ["get_recent_IERS_A_table"]
+
+def get_recent_IERS_A_table(warn_update=7*u.day):
+    """
+    Update the IERS Bulletin A table for accurate astropy calculations near
+    the present time.
+
+    The IERS A table records recent measurements and near-future predictions of
+    the Earth's orientation. A warning will be raised if the tables used are
+    more than ``warn_update`` old.
+    """
+    local_iers_a_path = get_pkg_data_filename('data/iers_bulletin_a.txt')
+    table = iers.IERS_A.open(local_iers_a_path)
+
+    # Use polar motion flag to identify last observation before predictions
+    index_of_last_observation = ''.join(table['PolPMFlag_A']).index('IP')
+    time_of_last_observation = Time(table['MJD'][index_of_last_observation],
+                                    format='mjd')
+    time_since_last_update = Time.now() - time_of_last_observation
+
+    # If the IERS bulletin is more than 7 days old, try to download a fresh one.
+    # Note that this doesn't ensure that the file that you download will be
+    # newer, it just ensures that you've tried to download the latest copy.
+    if warn_update < time_since_last_update:
+        try:
+            path_to_tmp_file = download_file(iers.IERS_A_URL, cache=False,
+                                             show_progress=False)
+            os.remove(local_iers_a_path)
+            os.rename(path_to_tmp_file, local_iers_a_path)
+            table = iers.IERS_A.open(local_iers_a_path)
```

Comment:
This is dangerous.  If I understand right, you're writing the new IERS file into the *source* directory.  In many situations (i.e., just about any linux system-level install) this will not work because the user's permissions are set to not allow users to alter the source (after all, that's a major security risk if a malicious piece of code edits a file to do ``os.system('rm -rf /')``!), and that would include the `data` directory.

So I would instead suggest doing this:
```python
from astropy.config.paths import get_cache_dir
path_to_write_to = os.path.join(get_cache_dir(), 'astroplan', 'iers_bulletin_a.txt')
```
You'll have to add a bit more logic to first check the data dir with this approach, but perhaps the easiest way is to *always* use the `path_to_write_to` as I define above as the file to read from, and do something like "if the file is absent, copy over what's in `get_pkg_data_filename`'.  Note this will also make the approach I'm about to describe in the comments a bit easier.